pkg/agent/cniserver/server.go

```diff
@@ -461,6 +467,22 @@ func (s *CNIServer) CmdAdd(ctx context.Context, request *cnipb.CniCmdRequest) (*
 	klog.Infof("CmdAdd for container %v succeeded", cniConfig.ContainerId)
 	// mark success as true to avoid rollback
 	success = true
+
+	if s.secondaryNetworkEnabled {
+		// Delete existing CNI information (if present) in the cache specific to the PodName and PodNamespace
+		prevCNIInfo := s.podConfigurator.podInfoStore.GetCNIConfigInfoPerPod(podName, podNamespace)
+		if len(prevCNIInfo) > 0 {
+			for _, podInfo := range prevCNIInfo {
+				s.podConfigurator.podInfoStore.DeleteCNIConfigInfo(podInfo)
+			}
+		}
+		// Go cache the CNI server info at CNIConfigInfo cache, for podWatch usage
+		cniInfo := &cnipodcache.CNIConfigInfo{CNIVersion: cniVersion, PodName: podName, PodNameSpace: podNamespace,
+			ContainerID: cniConfig.ContainerId, ContainerNetNS: netNS, ContainerIfDev: cniConfig.Ifname,
+			MTU: cniConfig.MTU}
+		s.podConfigurator.podInfoStore.AddCNIConfigInfo(cniInfo)
```

Comment:
@arunvelayutham @jianjuns is this the solution we agreed to? Is it not possible that the cache information will be deleted for the previous Pod before we have had a chance to process the Pod Delete event on the controller?

I thought we were considering the 2 following solutions (in addition to leveraging `.Status.PodIP` of course):
1) short term: if there are more than 2 matching entries in the cache when processing a Pod Add event, requeue the event
2) long term: trigger clean-up operations when we process the CNI Del event instead of using the Pod Delete event in the controller

Reply:
Right. I think we concluded to do approach 1 for this PR, and so we should not delete the existing cache entry in this func.